### PR TITLE
Alex/error handling

### PR DESCRIPTION
**Pull request by AlexLam1997**

### CODE REVIEW

File: backend/routes/users.js [hunk 1]
```diff
@@ -71,12 +47,10 @@ router.post("/completedCourses/comparison", function(req, res) {
...
  */
-router.get("/getCompletedCourses", async (req, res) => {
+router.get("/getCompletedCourses", asyncMiddleware(async (req, res) => {
   const student_id = req.query.studentID;
   const courses = await users.getCompletedCourses(student_id);
-  res.status(200).send(courses);
-});
-
-
+  res.status(200).send(customResponse(courses, null));
```
Comment (KevenLYF) [inline backend/routes/users.js]: If you are using undefined in customReponse.js  shouldn't you neglect the error field here ?

**Comment by KevenLYF** [comment]
Overall looks awesome, this is a really good example for handling errors using Express middleware, don't know if we can get bonus point of it :) . Now I think we should focus more on the features.

File: backend/app.js [hunk 2]
```diff
@@ -1,23 +1,33 @@
...
 app.use('/courses', coursesRouter);
+app.use(function (err, req, res, next) {
+    console.error(err.stack)
+    if (!isNaN(err.code)) {
+        res.status(err.code)
+    } else {
+        res.status(500)
+    }
+    res.send(customResponse('An error occured', err))
+})
```
Comment (Ty-Won) [inline backend/app.js]: What are you handing over here?

File: backend/routes/users.js (hunk 1 above)
Comment (AlexLam1997) [reply backend/routes/users.js]: oh i didnt know that would work, ill fix and see what it gives me

File: backend/app.js (hunk 2 above)
Comment (KevenLYF) [reply backend/app.js]: It's an Express middleware to catch and handle errors. The errors are passed from next

Comment (AlexLam1997) [reply backend/app.js]: The first argument is what goes in the "result" field of the response. Second is only if an error occurs

File: backend/test/users_test.js [hunk 3]
```diff
@@ -190,52 +189,53 @@ describe("Tests add student user", async function () {
...
+  //       "I.Have.A.Really.Long.Email.Address@emailDomainToo.com",
+  //       "535235231"
+  //     )
+  //     .then(response => {
+  //       return new Promise(function (resolve) {
+  //         assert.equal(response, "true");
+  //         resolve();
+  //       }).then(done);
+  //     });
+  // });
```
Comment (si-x-li) [inline backend/test/users_test.js]: No need to comment out tests. Mocha allows you to skip them using .skip

Comment (AlexLam1997) [reply backend/test/users_test.js]: Oh interesting, ill do that instead thanks!

**Pull request merged**